### ISSUE TITLE
can not copy and paste passwords in to Jitsi on Linux

### ISSUE THREAD

**Issue by aoighost** [jitsi/jitsi#125]
On Linux, you can not paste credentials in to Jitsi for things like XMPP accounts. This appears to be caused by not having a signed applet with proper settings to allow accessing the clipboard: https://stackoverflow.com/questions/8513740/copy-paste-not-working-in-a-signed-applet

**Comment by damencho** [comment]
Jitsi is not an applet. Please use the users mailinglist.

**Comment by vith** [comment]
Found this report on the mailing list http://lists.jitsi.org/pipermail/users/2014-April/006968.html

Pasting from KeePass still doesn't work. Copying from a terminal and pasting does work.

I'm on Arch Linux.
Kernel: `4.2.5-1-lqx #1 ZEN SMP PREEMPT Thu Oct 29 11:58:47 CDT 2015 x86_64 GNU/Linux`
Jitsi: `aur/jitsi 2.8.5426-2`
Oracle java: `aur/jdk 8u66-1`
Desktop environment: `extra/gnome-shell 3.18.1-2`
KeePass: `community/keepass 2.30-1`

```
Atom was 0
23:56:06.562 SEVERE: [39] util.UtilActivator.uncaughtException().108 An uncaught exception occurred in thread=Thread[AWT-EventQueue-0,6,main] and message was: Failed to retrieve atom name.
java.lang.NullPointerException: Failed to retrieve atom name.
	at sun.awt.X11.XlibWrapper.XGetAtomName(Native Method)
	at sun.awt.X11.XAtom.getName(XAtom.java:186)
	at sun.awt.X11.XDataTransferer.getTargetNameForAtom(XDataTransferer.java:165)
	at sun.awt.X11.XDataTransferer.getNativeForFormat(XDataTransferer.java:154)
	at sun.awt.datatransfer.DataTransferer.getFlavorsForFormats(DataTransferer.java:736)
	at sun.awt.datatransfer.ClipboardTransferable.<init>(ClipboardTransferable.java:90)
	at sun.awt.X11.XClipboard.getContents(XClipboard.java:108)
	at javax.swing.TransferHandler$TransferAction.actionPerformedImpl(TransferHandler.java:1755)
	at javax.swing.TransferHandler$TransferAction.access$800(TransferHandler.java:1692)
	at javax.swing.TransferHandler$TransferAction$1.run(TransferHandler.java:1715)
	at javax.swing.TransferHandler$TransferAction$1.run(TransferHandler.java:1713)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at javax.swing.TransferHandler$TransferAction$2.run(TransferHandler.java:1730)
	at javax.swing.TransferHandler$TransferAction$2.run(TransferHandler.java:1728)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.TransferHandler$TransferAction.actionPerformed(TransferHandler.java:1727)
	at javax.swing.text.JTextComponent.invokeAction(JTextComponent.java:1462)
	at javax.swing.text.JTextComponent.paste(JTextComponent.java:1440)
	at javax.swing.text.DefaultEditorKit$PasteAction.actionPerformed(DefaultEditorKit.java:1363)
	at javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1663)
	at javax.swing.JComponent.processKeyBinding(JComponent.java:2882)
	at javax.swing.JComponent.processKeyBindings(JComponent.java:2929)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2845)
	at java.awt.Component.processEvent(Component.java:6312)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:806)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1074)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:945)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:771)
	at java.awt.Component.dispatchEventImpl(Component.java:4762)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by ibauersachs** [comment]
There is no code of Jitsi being executed in the stacktrace you pasted. You might want to report this bug against Java, but given that Keepass doesn't use the normal clipboard but rather a mixture of inter-window keystroke sending and the clipboard I'd say that it's a Keepass problem.

**Comment by vith** [comment]
> ...but given that Keepass doesn't use the normal clipboard but rather a mixture of inter-window keystroke sending and the clipboard I'd say that it's a Keepass problem.

I'm not using the "auto-type" feature, just copying to clipboard.

> You might want to report this bug against Java,...

Found it, open since 2005: http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6322854

Same problem in NetBeans, they're just catching the exception but that doesn't make the paste succeed: https://netbeans.org/bugzilla/show_bug.cgi?id=126145

KeePass is using xsel to copy to clipboard, which has an unreleased patch to avoid this problem: https://github.com/kfish/xsel/pull/6

I was able to reproduce the issue directly with `echo "asdf" | xsel -i -b`.

Oh well.
